lib/src/install/config.rs

```diff
@@ -47,6 +47,11 @@ pub(crate) struct InstallConfiguration {
     /// Kernel arguments, applied at installation time
     #[serde(skip_serializing_if = "Option::is_none")]
     pub(crate) kargs: Option<Vec<String>>,
+    /// Architecture of the current system (inaccessible to users)
+    #[serde(skip_deserializing)]
+    pub(crate) detected_arch: String,
```

Comment:
It's confusing that this is part of the structure, even if it's not supposed to be visible to users.  I think we can make it ambient data passed to the merge function instead, although I know that's more code churn.

Reply:
My only concern with passing ambient data to the merge function is the fact that we need to add another parameter to it. The merge function is used for other structs outside of `InstallConfiguration` (ie. `RootFS` and `BasicFilesystems`). However, if we're okay with changing all instances in which the `merge` function is called, I can make implement this change.

Reply:
Fixed by passing an EnvProperties struct to the merge function: https://github.com/containers/bootc/pull/567/files#diff-e592ffb0075409e77a854e0502f4f5ffa7ac34d564042381036993cfa28c8309R11-R16